src/pybind/mgr/dashboard/frontend/e2e/cluster/osds.e2e-spec.ts

```diff
@@ -32,4 +33,51 @@ describe('OSDs page', () => {
       expect(osds.getTabText(1)).toEqual('Overall Performance');
     });
   });
+
+  describe('check existence of fields on OSD page', () => {
+    it('should check that number of rows and count in footer match', () => {
+      osds.navigateTo();
+      osds
+        .getTableCount()
+        .getText()
+        .then((tableCount) => {
+          tableCount = tableCount.slice(13, -6); // Grabs number of hosts from table footer
+          expect(osds.getTableRows().count()).toMatch(tableCount);
+        });
+    });
+
+    it('should verify that selected footer increases when an entry is clicked', () => {
+      osds.navigateTo();
+      osds.getFirstCell().click(); // clicks first osd
+      osds
+        .getTableCount()
+        .getText()
+        .then((tableCount) => {
+          tableCount = tableCount.substring(0, 1); // Grabs number of hosts from table footer
+          expect(tableCount).toMatch('1');
+        });
+    });
+
+    it('should verify that buttons exist', () => {
+      osds.navigateTo();
+      expect(element(by.cssContainingText('button', 'Scrub')).isPresent()).toBe(true);
+      expect(
+        element(by.cssContainingText('button', 'Cluster-wide configuration')).isPresent()
+      ).toBe(true);
+    });
+
+    it('should check the number of tabs when selecting an osd is correct', () => {
+      osds.navigateTo();
+      osds.getFirstCell().click(); // clicks first osd
+      expect(osds.getTabsCount()).toEqual(7); // includes tabs at the top of the page
```

Comment:
As mentioned above please consider to observe the content in order to make a valid e2e test and not a unit test in an e2e framework.

As a short explanation:
1. Unit tests are much faster and you get a good traceback to work with.
2. E2E are slower but test the integration between front and back end (on a real system). This means they have to make sure to test if the retrieved data from the backend is presented as planned through the frontend. This will catch unforeseen changes between backend and frontend.